### PR TITLE
Feature/tao 6034/encryption data

### PR DESCRIPTION
**Pull request by ionutpad**
The instruction how to setup can be found in the readme. (Server vs Client scripts)

Depends on:
- [x] https://github.com/oat-sa/generis/pull/480
- [x] https://github.com/oat-sa/extension-tao-delivery-rdf/pull/181
- [x] https://github.com/oat-sa/tao-core/pull/1674
- [x] https://github.com/oat-sa/extension-tao-proctoring/pull/561
- [x] https://github.com/oat-sa/extension-tao-testtaker/pull/81
- [x] https://github.com/oat-sa/extension-tao-sync/pull/5

### CODE REVIEW

File: model/Service/DeliveryMonitoring/EncryptedMonitoringService.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{
+    const TEST_TAKER_LOGIN = 'test_taker_login';
+    /**
+     * @param \oat\taoProctoring\model\monitorCache\implementation\DeliveryMonitoringData $data
+     * @param \oat\oatbox\user\User $user
+     * @return \oat\taoProctoring\model\monitorCache\implementation\DeliveryMonitoringData
+     */
+    protected function updateTestTakerInformation($data, $user)
+    {
+        $login = $user->getPropertyValues(GenerisRdf::PROPERTY_USER_LOGIN);
```
Comment (siwane) [inline model/Service/DeliveryMonitoring/EncryptedMonitoringService.php]: Use getOnePropertyValue or getUniquePropertyValue

File: model/Service/EncryptionSymmetricServiceHelper.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * @param string $key
+     * @return EncryptionSymmetricService
+     * @throws \Exception
+     */
+    public function getEncryptionService($key)
+    {
+        if (is_null($this->encryptionService)) {
+            /** @var EncryptionSymmetricService $service */
+            $service = $this->getServiceLocator()->get($this->getOption(static::OPTION_ENCRYPTION_SERVICE));
```
Comment (siwane) [inline model/Service/EncryptionSymmetricServiceHelper.php]: This sound like a magic for me, do u want u to replace it by an abstract method?

File: manifest.php [hunk 3]
```diff
@@ -52,6 +55,8 @@
             RegisterEncryptionSymmetricService::class,
             RegisterEncryptResultStorage::class,
             RegisterDecryptResultStorage::class,
+            RegisterSimpleKeyProviderService::class,
```
Comment (siwane) [inline manifest.php]: This one is not usefull, SimpleKeyProvider have not options and FileKeyProvider will override it

File: model/Service/Result/EncryptResultService.php [hunk 4]
```diff
@@ -23,10 +23,13 @@
 use common_persistence_KvDriver;
 use oat\oatbox\log\LoggerAwareTrait;
 use oat\oatbox\service\ConfigurableService;
+use oat\taoDelivery\model\execution\DeliveryExecutionInterface;
+use oat\taoDelivery\model\execution\ServiceProxy;
 use oat\taoResultServer\models\Entity\ItemVariableStorable;
 use oat\taoEncryption\Service\EncryptionServiceInterface;
 use oat\taoResultServer\models\Entity\TestVariableStorable;
 use oat\taoResultServer\models\Entity\VariableStorable;
+use oat\taoSync\model\result\DetectTestAndItemIdentifiers;
```
Comment (siwane) [inline model/Service/Result/EncryptResultService.php]: There is no dependency from taoEncryption to taoSync

File: scripts/tools/SetupUserSynchronizer.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+        /** @var UserSynchronizer $proctor */
+        $proctor = $syncOptions[SyncService::OPTION_SYNCHRONIZERS]['proctor'];
+        $proctor->setOption(UserSynchronizer::OPTIONS_FORMATTER_CLASS, EncryptUserSyncFormatter::SERVICE_ID);
+
+        /** @var UserSynchronizer $testTaker */
+        $testTaker = $syncOptions[SyncService::OPTION_SYNCHRONIZERS]['test-taker'];
+        $testTaker->setOption(UserSynchronizer::OPTIONS_FORMATTER_CLASS, EncryptUserSyncFormatter::SERVICE_ID);
+
+        $syncService->setOptions(array_merge(
```
Comment (siwane) [inline scripts/tools/SetupUserSynchronizer.php]: you don't need to set options, you have already done

File: scripts/update/Updater.php [hunk 6]
```diff
@@ -49,5 +51,17 @@ public function update($initialVersion)
             );
             $this->setVersion('0.5.0');
         }
+
+        if ($this->isVersion('0.5.0')){
+            $simpleKeyProvider = new SimpleKeyProviderService([]);
+
+            $this->getServiceManager()->register(SimpleKeyProviderService::SERVICE_ID, $simpleKeyProvider);
+
+            $fileKeyProvider = new FileKeyProviderService([]);
```
Comment (siwane) [inline scripts/update/Updater.php]: I think your forgot the filesystem option

File: scripts/update/Updater.php [hunk 7]
```diff
@@ -49,5 +51,17 @@ public function update($initialVersion)
             );
             $this->setVersion('0.5.0');
         }
+
+        if ($this->isVersion('0.5.0')){
+            $simpleKeyProvider = new SimpleKeyProviderService([]);
```
Comment (siwane) [inline scripts/update/Updater.php]: Not needed if FileKeyProviderService is registered after

File: model/Service/DeliveryMonitoring/EncryptedMonitoringService.php (hunk 1 above)
Comment (ionutpad) [reply model/Service/DeliveryMonitoring/EncryptedMonitoringService.php]: no necessary

File: manifest.php (hunk 3 above)
Comment (ionutpad) [reply manifest.php]: no necessary

File: scripts/tools/SetupUserSynchronizer.php (hunk 5 above)
Comment (ionutpad) [reply scripts/tools/SetupUserSynchronizer.php]: no necessary

File: scripts/update/Updater.php (hunk 7 above)
Comment (ionutpad) [reply scripts/update/Updater.php]: no necessary

**Pull request merged**